src/storage/src/storage_state.rs

```diff
@@ -255,7 +255,7 @@ pub struct StorageState {
     /// and we should aim for that but are not there yet.
     pub source_uppers: BTreeMap<GlobalId, Rc<RefCell<Antichain<mz_repr::Timestamp>>>>,
     /// Handles to created sources, keyed by ID
-    pub source_tokens: BTreeMap<GlobalId, Rc<dyn Any>>,
+    pub source_tokens: BTreeMap<GlobalId, Vec<PressOnDropButton>>,
```

Comment:
Maybe put a comment here and on sink_tokens about this needing to stay as `PressOnDropButton` so someone doesn't unknowingly regress this later

Reply:
Good idea